lib/pbench/test/unit/server/query_apis/test_iteration_samples.py

```diff
@@ -0,0 +1,317 @@
+import pytest
+from http import HTTPStatus
+from pbench.server.api.resources.query_apis.iteration_samples import IterationSampleRows
+from pbench.test.unit.server.headertypes import HeaderTypes
+from pbench.test.unit.server.query_apis.commons import Commons
+
+
+class TestIterationSamplesRows(Commons):
+    """
+    Unit testing for IterationSamplesRows class.
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    @pytest.fixture(autouse=True)
+    def _setup(self, client):
+        super()._setup(
+            cls_obj=IterationSampleRows(client.config, client.logger),
+            pbench_endpoint="/dataset/samples/namespace",
+            elastic_endpoint="/_search",
+            payload={"run_id": "random_md5_string1"},
+        )
+
+    def test_query(
+        self,
+        server_config,
+        query_api,
+        user_ok,
+        pbench_token,
+        build_auth_header,
+        find_template,
+        provide_metadata,
+    ):
+        """
+        Check the construction of Elasticsearch query URI and filtering of the
+        response body. Note that the mock set up by the attach_dataset fixture
+        matches the dataset name to the dataset's owner.
+        """
+        response_payload = {
```

Comment:
The way we've got this structured, playing with different ES payloads in the unit tests is awkward. I haven't really come up with a better way. In any case, there was a discussion in another context about dropping fields (that appear in the document template) which have no values in this specific dataset ... are we testing that case? 😁

Reply:
Yes the mappings contain other properties like `timestamp_original` that has a type keyword but they are not included in the aggregation only `"run", "sample", "iteration", "benchmark"` fields are used

Reply:
OK, that wasn't helpful; but it seems that your sample Elasticsearch response does show empty buckets, and your response comparison would fail if they were present, so I think this is good.